stock_lot_suffix/models/stock_move.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _action_done(self, cancel_backorder=False):
+        res = super()._action_done(cancel_backorder=cancel_backorder)
+        for move in self:
+            if not move.lot_ids:
```

Comment:
I think the original logic may have had an issue (looks like non-purchase related moves would wipe out the `channel_category` value of the lot). @nobuQuartile Can you please help confirm this point?
```suggestion
            if not move.purchase_line_id or not move.lot_ids:
```

Reply:
@yostashiro 
You are correct. I have verified in my development environment that non-purchase-related moves set the channel_category value of the lot to false.